rpc/v2_endpoints.go

```diff
@@ -0,0 +1,367 @@
+package rpc
+
+import (
+	"context"
+	"encoding/hex"
+	"github.com/pkg/errors"
+	"github.com/qubic/go-archiver/protobuff"
+	"github.com/qubic/go-archiver/store"
+	"github.com/qubic/go-node-connector/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func (s *Server) GetLatestTickV2(ctx context.Context, req *emptypb.Empty) (*protobuff.GetLatestTickResponseV2, error) {
+	return s.GetLatestTick(ctx, req)
+}
+
+func (s *Server) GetTickV2(ctx context.Context, req *protobuff.GetTickRequestV2) (*protobuff.GetTickResponseV2, error) {
+
+	//TODO: Implement
+
+	return nil, status.Errorf(codes.Unimplemented, "Not yet implemented")
+}
+
+func (s *Server) GetTickQuorumDataV2(ctx context.Context, req *protobuff.GetTickRequestV2) (*protobuff.GetQuorumTickDataResponse, error) {
+	return s.GetQuorumTickData(ctx, &protobuff.GetQuorumTickDataRequest{
+		TickNumber: req.TickNumber,
+	})
+}
+
+func (s *Server) GetTickChainHashV2(ctx context.Context, req *protobuff.GetTickRequestV2) (*protobuff.GetChainHashResponse, error) {
+	return s.GetChainHash(ctx, &protobuff.GetChainHashRequest{
+		TickNumber: req.TickNumber,
+	})
+}
+
+func (s *Server) GetTickStoreHashV2(ctx context.Context, req *protobuff.GetTickRequestV2) (*protobuff.GetChainHashResponse, error) {
+	return s.GetStoreHash(ctx, &protobuff.GetChainHashRequest{
+		TickNumber: req.TickNumber,
+	})
+}
+
+func (s *Server) GetTickTransactionsV2(ctx context.Context, req *protobuff.GetTickTransactionsRequestV2) (*protobuff.GetTickTransactionsResponseV2, error) {
+
+	tickRequest := &protobuff.GetTickRequestV2{
```

Comment:
it's better to define the request as a value and the pass the reference to each function/method, this way we know 100% that the variable will not end on the heap